### PR TITLE
API 2.0 FIT OS bootstrap test for Linux distros

### PR DESCRIPTION
**Pull request by gpaulos**
New API 2.0 FIT OS bootstrap test for Linux distros. Payload can be specified by loading a file using the -extra parameter. Examples in the file header comments.

There is a default payload for CentOS 7, but the repo spec is, as always, site specific. There is no generic repo spec that will work in all situations unless some conventions are adopted. Erika and I are unsure whether there should be a default payload at all. Any thoughts?

This is the base test script. It is intended to be enhanced with additional tests to verify payload parameters in future stories. 

@hohene @johren @derrickostertag @stuart-stanley @jimturnquist

### CODE REVIEW

**Comment by johren** [comment]
test this please

File: test/tests/bootstrap/test_api20_linux_bootstrap.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                                      '/workflows',
+                                      action='post', payload=PAYLOAD)
+        if result['status'] == 201:
+            # workflow running
+            log.info_5(" TaskID: " + result['json']['instanceId'])
+            log.info_5(" Payload: " + fit_common.json.dumps(PAYLOAD))
+            workflowid = result['json']['instanceId']
+        else:
+            # workflow failed with response code
+            log.error(" TaskID: " + result['text'])
```
Comment (johren) [inline test/tests/bootstrap/test_api20_linux_bootstrap.py]: Is this really a TaskID?

File: test/tests/bootstrap/test_api20_linux_bootstrap.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+
+@attr(all=False)
+class api20_bootstrap_linux(fit_common.unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Get the list of nodes
+        NODECATALOG = fit_common.node_select()
+        # Select one node at random
+        cls.__NODE = NODECATALOG[random.randint(0, len(NODECATALOG) - 1)]
```
Comment (gpaulos) [inline test/tests/bootstrap/test_api20_linux_bootstrap.py]: Need to specify a node ID or node index when running in parallel.

**Comment by gpaulos** [comment]
Requested changes pushed up.

**Comment by johren** [comment]
Paired with @hohene

**Pull request merged**